pennylane/qnode.py

```diff
@@ -605,8 +605,25 @@ def __call__(self, *args, **kwargs):
             # construct the tape
             self.construct(args, kwargs)
 
-        # execute the tape
-        res = self.qtape.execute(device=self.device)
+        # Execute the tape.
+        # If the observable contains a Hamiltonian and the device does not
+        # support Hamiltonians, split tape into multiple tapes and execute
+        # sequentially. In future, this logic should be moved to the device
+        # to allow for more efficient batch execution.
+        supports_hamiltonian = self.device.supports_observable("Hamiltonian")
+        hamiltonian_in_obs = "Hamiltonian" in [obs.name for obs in self.qtape.observables]
+        if hamiltonian_in_obs and (self.device.shots is not None or not supports_hamiltonian):
+            try:
+                tapes, fn = qml.transforms.hamiltonian_expand(self.qtape, group=False)
+            except ValueError as e:
+                raise ValueError(
+                    "At the moment, only single expectations of Hamiltonian observables can be returned"
+                    "on the {} device.".format(self.device.name)
+                ) from e

```

Comment:
I have to admit, I don't quite follow this error message. Does it catch the case where someone tries to do `expval(H1), expval(H2)`?

Or is it more general than that, only allowing `expval(H)` and no other measurements (e.g., `expval(H), expval(PauliZ(0))`)?

Reply:
Ah, I see from lower down that _only_ `expval(H)` is allowed, no other measurements.

Reply:
At least in this very first iteration!